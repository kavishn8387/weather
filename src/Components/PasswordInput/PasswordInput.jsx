import { InputAdornment, IconButton, TextField } from '@material-ui/core';
import React, { Component } from 'react';
import Visibility from '@material-ui/icons/Visibility';
import VisibilityOff from '@material-ui/icons/VisibilityOff';

class PasswordInput extends Component {
    constructor(props) {
        super(props);
        this.state = {
            showPassword: false,
            password: ''
        };
    }

    handleClickShowPassword = () => {
        this.setState((prevState) => ({
            showPassword: !prevState.showPassword,
        }));
    };

    handleChangePassword = (event) => {
        const password = event.target.value;
        if (password.split(' ').length > 1) {
            return false;
        }
        this.setState({ password }, () => {
            this.props.handleChangePassword(password);
        });
        return true;
    }

    render() {
        const { style, label, error, onKeyUp } = this.props;
        const { showPassword, password } = this.state;

        return (
            <TextField
                {...error}
                fullWidth
                className={style}
                label={label}
                type={showPassword ? 'text' : 'password'}
                variant="outlined"
                value={password}
                onChange={this.handleChangePassword}
                onKeyUp={onKeyUp}
                placeholder="****************"
                InputProps={{
                    endAdornment: (
                        <InputAdornment position="end">
                            <IconButton
                                id="one"
                                aria-label="Toggle password visibilty"
                                onClick={this.handleClickShowPassword}
                            >
                                {showPassword ? (
                                    <Visibility />
                                ) : (
                                    <VisibilityOff />
                                )}
                            </IconButton>
                        </InputAdornment>
                    ),
                    inputProps: {
                        pattern: '^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[#?!@$%^&*-]).{8,}$'
                    }
                }}
            />
        );
    }
}

export default PasswordInput;
