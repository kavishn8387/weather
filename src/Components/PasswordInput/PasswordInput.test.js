import React from 'react';
import Enzyme, { shallow, enzyme, configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { withRouter } from 'react-router';
import { IconButton } from '@material-ui/core/IconButton';

import PasswordInput from './PasswordInput';

Enzyme.configure({ adapter: new Adapter() });

describe('<PasswordInput />', () => {
    it('renders', () => {
        const wrapper = shallow(<PasswordInput />);
        expect(wrapper).toMatchSnapshot();
    });


    it('Test click event on IconButton', () => {
        const component = mount(<PasswordInput />);
        component.find('IconButton').simulate('click');
        expect(component.state().showPassword).toBe(true);
    });
});
