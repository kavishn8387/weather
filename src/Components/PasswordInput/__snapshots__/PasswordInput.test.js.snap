// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PasswordInput /> renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PasswordInput />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "InputProps": Object {
        "endAdornment": <WithStyles(WithFormControlContext(InputAdornment))
          position="end"
        >
          <WithStyles(IconButton)
            aria-label="Toggle password visibilty"
            id="one"
            onClick={[Function]}
          >
            <pure(VisibilityOffIcon) />
          </WithStyles(IconButton)>
        </WithStyles(WithFormControlContext(InputAdornment))>,
        "inputProps": Object {
          "pattern": "^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[#?!@$%^&*-]).{8,}$",
        },
      },
      "className": undefined,
      "fullWidth": true,
      "label": undefined,
      "onChange": undefined,
      "onKeyUp": undefined,
      "placeholder": "****************",
      "required": false,
      "select": false,
      "type": "password",
      "value": undefined,
      "variant": "outlined",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "InputProps": Object {
          "endAdornment": <WithStyles(WithFormControlContext(InputAdornment))
            position="end"
          >
            <WithStyles(IconButton)
              aria-label="Toggle password visibilty"
              id="one"
              onClick={[Function]}
            >
              <pure(VisibilityOffIcon) />
            </WithStyles(IconButton)>
          </WithStyles(WithFormControlContext(InputAdornment))>,
          "inputProps": Object {
            "pattern": "^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[#?!@$%^&*-]).{8,}$",
          },
        },
        "className": undefined,
        "fullWidth": true,
        "label": undefined,
        "onChange": undefined,
        "onKeyUp": undefined,
        "placeholder": "****************",
        "required": false,
        "select": false,
        "type": "password",
        "value": undefined,
        "variant": "outlined",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
